introspection-engine/introspection-engine-tests/tests/re_introspection/vitess.rs

```diff
@@ -403,22 +403,23 @@ async fn referential_actions_are_kept_intact(api: &TestApi) -> TestResult {
         }
 
         model B {
-          id  Int @id @default(autoincrement())
-          aId Int
-          a   A   @relation(fields: [aId], references: [id], onDelete: SetNull)
+          id  Int  @id @default(autoincrement())
+          aId Int?
+          a   A?   @relation(fields: [aId], references: [id], onDelete: SetNull)
         }
     "#};
 
+    // note: we've probably introduced a regression here, `onDelete: SetNull` is gone from B's relation.
     let expected = expect![[r#"
         model A {
           id Int @id @default(autoincrement())
           bs B[]
         }
 
         model B {
-          id  Int @id @default(autoincrement())
-          aId Int
-          a   A   @relation(fields: [aId], references: [id], onDelete: SetNull)
+          id  Int  @id @default(autoincrement())
+          aId Int?
+          a   A?   @relation(fields: [aId], references: [id])
         }
```

Comment:
**Warning**: this indicates a regression in re-introspection: by simply making `B.aId` optional, the referential action `onDelete: SetNull` is now gone.

The original `Vitess` re-introspection test CI output can be seen [here](https://github.com/prisma/prisma-engines/pull/3306).

Reply:
Uhm, it seems it comes from [`dml::field::RelationField::default_on_delete_action`](https://github.com/prisma/prisma-engines/blob/main/libs/dml/src/field.rs#L375-L383).

```rust
    pub fn default_on_delete_action(&self) -> ReferentialAction {
        use ReferentialAction::*;

        match self.referential_arity {
            FieldArity::Required if self.supports_restrict_action.unwrap_or(true) => Restrict,
            FieldArity::Required => NoAction,
            _ => SetNull, // we match this case, 
        }
    }
```

which causes this condition in [`dml::render::render_datamodel::render_field_attributes`](https://github.com/prisma/prisma-engines/blob/main/libs/dml/src/render/render_datamodel.rs#L271-L275) to trigger: 

```rust
    if let Some(ref_action) = relation_info.on_delete {
        if rf.default_on_delete_action() != ref_action { // this condition becomes false
            args.push((Some("onDelete"), ref_action.to_string()));
        }
    }
```

Reply:
Since the test above explicitly wants to verify that referential actions are preserved during re-introspection, I have swapped `SetNull` with `Restrict`.

Reply:
`SetNull` is the default action, which we do not render.